beginner_source/onnx/onnx_registry_tutorial.py

```diff
@@ -0,0 +1,469 @@
+# -*- coding: utf-8 -*-
+
+"""
+`Introduction to ONNX <intro_onnx.html>`_ ||
+`Export a PyTorch model to ONNX <export_simple_model_to_onnx_tutorial.html>`_ ||
+**Introduction to ONNX Registry**
+
+Introduction to ONNX Registry
+=============================
+
+**Authors:** Ti-Tai Wang (titaiwang@microsoft.com)
+"""
+
+
+###############################################################################
+# Overview
+# ~~~~~~~~
+#
+# This tutorial is an introduction to ONNX registry, which
+# empowers users to create their own ONNX registries enabling
+# them to address unsupported operators in ONNX.
+#
+# In this tutorial, we will cover the following scenarios:
+#
+# * Unsupported ATen operators
+# * Unsupported ATen operators with existing ONNX Runtime support
+# * Unsupported PyTorch operators with no ONNX Runtime support
+#
+
+
+import torch
+print(torch.__version__)
+torch.manual_seed(191009)  # set the seed for reproducibility
+
+import onnxscript  # pip install onnxscript
+print(onnxscript.__version__)
+
+# NOTE: opset18 is the only version of ONNX operators we are
+# using in torch.onnx.dynamo_export for now.
+from onnxscript import opset18
+
+import onnxruntime  # pip install onnxruntime
+print(onnxruntime.__version__)
+
+
+######################################################################
+# Unsupported ATen operators
+# ---------------------------------
+#
+# ATen operators are implemented by PyTorch, and the ONNX exporter team must manually implement the
+# conversion from ATen operators to ONNX operators through [ONNX Script](https://onnxscript.ai/). Although the ONNX exporter
+# team has been making their best efforts to support as many ATen operators as possible, some ATen
+# operators are still not supported. In this section, we will demonstrate how you can implement any
+# unsupported ATen operators, which can contribute back to the project through the PyTorch GitHub repo.
```

Comment:
Actually, it's the third case that our users value the most, not the first one.

This paragraph is particularly serving scenario 1, and would be too specific for an overview. Instead, I think the intro below pretty much explains what we are going to do/learn in this tutorial.

```python
# This tutorial is an introduction to ONNX registry, which
# empowers users to create their own ONNX registries enabling
# them to address unsupported operators in ONNX.
```

The term "unsupported operators" unbiasly states that users should refer to here when they have "any kind" unsupported operator issue (could be unsupported python op / ATen op / total custom op).

Reply:
The additional paragraph to the tutorial was commented on the previous comment.

There I mentioned the text above was slightly modified to cover not only unsupported operators, but also replacing existing operators. It wasn't a copy and paste change only. it can still be improved as discussed in the original comment

Reply:
Replied in another comment